### PR TITLE
Subclassing refactoring

### PR DESCRIPTION
**Pull request by sdroege**
Supersedes https://github.com/gtk-rs/gtk-rs/pull/31 for now, I'll update that one to only include the `Value` changes once this one is merged.

TODO
  * [x] Implement a builder and a `fn signals() -> &'static [Signal]` method to `ObjectSubclass`

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]

**Review by bilelmoussaoui — APPROVED** [review]

**Comment by sdroege** [comment]
@bilelmoussaoui @GuillaumeGomez Can you review the newest commit too? That's doing the whole thing for signals now. I think that's a much nicer API, and it also gives a way for calling signals from subclasses more nicely.

**Comment by sdroege** [comment]
From my side this PR is ready, but before merging this I'd like to update all of GStreamer to not have it fail to build for too long. I'll probably spend Saturday or Sunday with that, it will take some time.

**Comment by sdroege** [comment]
https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/613 for the GStreamer side of things. Nice cleanup.

Now I just need to port over all the 50 or so subclasses in gst-plugins-rs :sob:

File: glib/src/subclass/types.rs [hunk 1]
```diff
@@ -128,6 +107,103 @@ pub unsafe trait IsImplementable<T: ObjectSubclass>: StaticType {
...
+                    interface_finalize: None,
+                    interface_data: ptr::null_mut(),
+                },
+            )
+        );
+        $types
+    });
+);
+
+interface_list_trait!(A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S);
```
Comment (GuillaumeGomez) [inline glib/src/subclass/types.rs:205]: I don't understand at all what this is about... What's the point of having this implemented randomly?

Comment (sdroege) [reply glib/src/subclass/types.rs:205]: What do you mean with randomly? This is implementing the `InterfaceList` trait on tuples of size 2 to 19

Comment (GuillaumeGomez) [reply glib/src/subclass/types.rs:205]: Exactly that. Do we really need to go up to 19 for example? Isn't it possible to make implementations on a trait that the tuples could implement?

Comment (sdroege) [reply glib/src/subclass/types.rs:205]: Well we could go up to any other number you would prefer. Implementing it not separately for each tuple size would require https://github.com/rust-lang/rfcs/issues/376 and until then a macro seemed nicer than duplicating this code a dozen times.

Comment (GuillaumeGomez) [reply glib/src/subclass/types.rs:205]: Oh damn, I was sure it was merged since the time... Then there is no issue.

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**